lldb/source/Plugins/TypeSystem/Swift/SwiftDemangle.h

```diff
@@ -60,6 +60,26 @@ NodeAtPath(swift::Demangle::NodePointer root,
   return ChildAtPath(root, kind_path.drop_front());
 }
 
+/// Find the first child node of root that satisfies cond.
+inline swift::Demangle::NodePointer
+FindIf(swift::Demangle::NodePointer root,
```

Comment:
This is kind of similar to `TypeSystemSwiftTypeRef::Contains`. I think you could just reuse that for this patch's use-case.